packages/flutter_tools/test/general.shard/android/android_workflow_test.dart

```diff
@@ -277,4 +277,35 @@ void main() {
     Stdio: () => stdio,
   });
 
+  testUsingContext('detects minimum required java version', () async {
+    final AndroidSdkVersion mockSdkVersion = MockAndroidSdkVersion();
+
+    // Mock a pass through scenario to reach _checkJavaVersion()
+    when(sdk.licensesAvailable).thenReturn(true);
+    when(sdk.platformToolsAvailable).thenReturn(true);
+    when(mockSdkVersion.sdkLevel).thenReturn(28);
+    when(mockSdkVersion.buildToolsVersion).thenReturn(Version(28, 0, 3));
+    when(sdk.sdkManagerPath).thenReturn('/foo/bar/sdkmanager');
+    when(sdk.latestVersion).thenReturn(mockSdkVersion);
+    when(sdk.validateSdkWellFormed()).thenReturn(<String>[]);
+
+    //Test with older version of JDK
+    when(processManager.run(argThat(contains('-version')))).thenAnswer((_) =>
+      Future<ProcessResult>.value(ProcessResult(0, 0, null, '1.7.0')));
+    final String errorMessage = userMessages.androidJavaMinimumVersion('1.7.0');
+
+    final ValidationResult validationResult = await AndroidValidator().validate();
+    expect(validationResult.type, ValidationType.partial);
+    expect(
+      validationResult.messages.last.message,
+      errorMessage,
+    );
+  }, overrides: <Type, Generator>{
+    AndroidSdk: () => sdk,
+    FileSystem: () => fs,
+    Platform: () => FakePlatform()..environment = <String, String>{'HOME': '/home/me', 'JAVA_HOME': 'home/java'},
```

Comment:
environment is usually an unmodifiable map. Its good to use one of these in the injection

```suggestion
    Platform: () => FakePlatform()..environment = Map<String, String>.unmodifiable(<String, String>{'HOME': '/home/me', 'JAVA_HOME': 'home/java'}),
```

Reply:
Should I make the same change for all the other tests in this suite?

Reply:
That would be wonderful, but you don't have to since you didn't touch those lines

Reply:
No trouble at all. There you go